src/modules/dashboard/DashboardEModeButton.tsx

```diff
@@ -40,14 +45,27 @@ export const DashboardEModeButton = ({
     <Trans>{getEmodeMessage(userEmodeCategoryId, baseAssetSymbol)}</Trans>
   );
 
+  useEffect(() => {
+    if (reserves.length !== reservesLength) {
+      const eModes: number[] = [];
+      reserves.forEach((reserve: ComputedReserveData) => {
+        if (eModes.indexOf(reserve.eModeCategoryId) === -1) {
+          eModes.push(reserve.eModeCategoryId);
+        }
+      });
+      setEmodes(eModes);
+      setReservesLength(reserves.length);
+    }
+  }, [reservesLength, setReservesLength, reserves]);
+
```

Comment:
Small optimization, but I think you could get rid of the useEffect and the state for reservesLength and eModes and just calculate the eModes on every render

Reply:
Removed this, the extra hook is not worth it since the component is re-rendered on reserves change anyways